analyses/transcriptomic-dimension-reduction/04-explore-sequencing-center-effects.Rmd

```diff
@@ -0,0 +1,433 @@
+  scale_alpha_manual(values = c(0.6, 0.8, 0.8)) +
+  ggpubr::theme_pubr() + 
+  guides(fill = FALSE, alpha = FALSE)
+```
+
+
+
+
+### Cancer group UMAP
+
+We can make this _even more_ finer-grained by exploring the distributions of cancer groups within this UMAP.
+Shapes denote the sequencing center, and colors denote the cancer group.
+
+```{r}
+ggplot(stranded_df) + 
+  aes(x = UMAP1, y = UMAP2, 
+      shape = seq_center, 
+      fill = cancer_group_display, 
+      color = seq_center, 
+      alpha = seq_center) + 
+  geom_point() + 
+  scale_shape_manual(values = c(21, 22, 24)) + 
+  scale_fill_manual(values  = pal_cg) +
+  scale_color_manual(values = c("grey70", "black", "black")) +
+  scale_alpha_manual(values = c(0.4, 0.6, 0.6)) +
+  ggpubr::theme_pubr() + 
+  guides(fill = FALSE, alpha = FALSE)
+```
+
+
+
+We can also make this plot only for cancer groups which were processed at >1 center:
+
+```{r}
+# first, which histologies are shared, aka processed at >1 center?
+shared_cg <- n_centers_cg %>%
+  dplyr::filter(n > 1,
+                RNA_library == "stranded") %>%
+  dplyr::distinct(cancer_group_display) %>%
+  dplyr::pull(cancer_group_display)
+
+
+stranded_df %>%
+  dplyr::filter(cancer_group_display %in% shared_cg) %>%
+  ggplot() + 
+  aes(x = UMAP1, y = UMAP2, 
+      shape = seq_center, 
+      fill = cancer_group_display, 
+      color = seq_center, 
+      alpha = seq_center) + 
+  geom_point() + 
+  scale_shape_manual(values = c(21, 22, 24)) + 
+  scale_fill_manual(values  = pal_cg) +
+  scale_color_manual(values = c("grey70", "black", "black")) +
+  scale_alpha_manual(values = c(0.6, 0.8, 0.8)) +
+  ggpubr::theme_pubr() + 
+  guides(fill = FALSE, alpha = FALSE)
+```
+
+
+## _Finer-grained_ cancer group UMAP
+
+First we'll set up with the other data frame, but we don't set the palette since there is no pre-defined palette here.
+```{r}
+# only keep stranded for the rest and set order for `seq_center` as group size
+stranded_df <- umap_df_full %>%
+  dplyr::filter(RNA_library == "stranded") %>%
+  dplyr::select(-RNA_library) %>%
+  dplyr::mutate(seq_center = forcats::fct_infreq(seq_center)) %>%
+  # arranging will help see points:
+  dplyr::arrange(seq_center)
+
+# How many cancer groups are there in total?
+total_cg <- unique(stranded_df$cancer_group)
+length(total_cg)
+
+# How many cancer groups _processed at >1 center_ are there?
+shared_cg_full <- n_centers_cg_full %>%
+  dplyr::filter(n > 1,
+                RNA_library == "stranded") %>%
+  dplyr::distinct(cancer_group) %>%
+  dplyr::pull(cancer_group)
+length(shared_cg_full)
+shared_cg_full
+```
+
+Since there are `r length(total_cg)` cancer groups, we'll focus on visualizing only the `r length(shared_cg_full)` cancer groups that are processed at >1 center.
+
+
+```{r}
+stranded_df_subset <- stranded_df %>%
+  dplyr::filter(cancer_group %in% shared_cg_full)
+
+# Legend does not render for cancer group colors properly, so create a hack legend first - 
```

Comment:
Something still looks off on the legend here. There is a purple color that doesn't seem to be present in the legend? Or maybe it is just the pink and a trick of the light? But there should be 9 cancer groups, right? and there are only 7 in the legend.